nautobot/dcim/forms.py

```diff
@@ -183,17 +183,21 @@ def clean(self):
 
         parent_field = "device" if "device" in self.cleaned_data else "virtual_machine"
         tagged_vlans = self.cleaned_data["tagged_vlans"]
+        mode = self.cleaned_data["mode"]
+
+        if mode != InterfaceModeChoices.MODE_TAGGED and tagged_vlans:
+            raise forms.ValidationError({"tagged_vlans": f"Clear tagged_vlans to set mode to {self.mode}"})
```

Comment:
Since `tagged_vlans` is auto cleared via Javascript whenever the mode is changed in UI, this is essentially redundant.